plugins/store/src/store.rs

```diff
@@ -174,142 +164,206 @@ impl<R: Runtime> StoreBuilder<R> {
+        let rid = self.app.resources_table().add_arc(store.clone());
+        stores.insert(self.path, rid);
+
+        Ok((store, rid))
+    }
+
+    /// Builds the [`Store`], also see [`build_or_existing`](Self::build_or_existing).
+    ///
+    /// This loads the store from disk and put the store in the app's resource table,
+    /// to remove it from the resource table, call [`Store::close_store`]
+    ///
+    /// # Errors
+    ///
+    /// If a store with this path is already in the resource table,
+    /// will return a [`crate::Error::AlreadyExists`]
     ///
     /// # Examples
     /// ```
     /// tauri::Builder::default()
     ///   .plugin(tauri_plugin_store::Builder::default().build())
     ///   .setup(|app| {
-    ///     let store = tauri_plugin_store::StoreBuilder::new(app, "store.json").build();
+    ///     let store = tauri_plugin_store::StoreBuilder::new(app, "store.json").new_or_existing()?;
     ///     Ok(())
     ///   });
     /// ```
-    pub fn build(self) -> Store<R> {
-        let collection = self.app.state::<StoreCollection<R>>();
-        let mut stores = collection.stores.lock().unwrap();
-        let store = stores
-            .get(&self.path)
-            .and_then(|store| store.upgrade())
-            .unwrap_or_else(|| {
-                let mut store = StoreInner::new(self.app.clone(), self.path.clone());
-                let _ = store.load(self.deserialize);
-                let store = Arc::new(Mutex::new(store));
-                stores.insert(
-                    self.path.clone(),
-                    Arc::<Mutex<StoreInner<R>>>::downgrade(&store),
-                );
-                store
-            });
-        drop(stores);
-        Store {
-            defaults: self.defaults,
-            serialize: self.serialize,
-            deserialize: self.deserialize,
-            auto_save: self.auto_save,
-            auto_save_debounce_sender: Arc::new(Mutex::new(None)),
-            store,
+    pub fn create(self) -> crate::Result<Arc<Store<R>>> {
+        let (store, _) = self.create_inner()?;
+        Ok(store)
+    }
+
+    pub(crate) fn create_inner(self) -> crate::Result<(Arc<Store<R>>, ResourceId)> {
+        let stores = self.app.state::<StoreState>().stores.clone();
+        self.build_inner(stores.lock().unwrap())
+    }
+
+    /// Get the existing store with the same path or creates a new [`Store`].
+    ///
+    /// If a store with the same path has already been loaded its instance is returned.
+    ///
+    /// See [`create`](Self::create) if you want to force create a new store in the path.
+    ///
+    /// # Examples
+    /// ```
+    /// tauri::Builder::default()
+    ///   .plugin(tauri_plugin_store::Builder::default().build())
+    ///   .setup(|app| {
+    ///     let store = tauri_plugin_store::StoreBuilder::new(app, "store.json").new_or_existing();
+    ///     Ok(())
+    ///   });
+    /// ```
+    pub fn new_or_existing(self) -> crate::Result<Arc<Store<R>>> {
```

Comment:
this can also be `build_new_or_load_existing` or simpler `build_new_or_existing`